pkgs/applications/science/programming/scyther/gui.nix

```diff
@@ -0,0 +1,43 @@
+{ fetchFromGitHub, python27Packages, graphviz, source, cli }:
+python27Packages.buildPythonApplication rec {
+  name = "scyther-gui-${source.rev}";
+  src = source;
+  
+  propagatedBuildInputs = [
+    cli
+    python27Packages.wxPython
+    graphviz
+  ];
+
+  format = "other";
+
+  phases = [ "unpackPhase" "patchPhase" "installPhase" "fixupPhase" "testPhase" ];
+
+  patchPhase = ''
```

Comment:
I'll fix that

And my reply to your comment about joining the outputs (can't reply to that for some reason): it separates stuff nicely I think, since the source needs to be shared by both parts and the gui depends on the cli with this setup one can only install the cli if desired. I'll think about it once more but I believe it's the cleanest like this. One other solution would be to just use 2 top-level packaged as you did before